.github/workflows/codeql-analysis.yml

```diff
@@ -18,8 +18,15 @@ on:
     # The branches below must be a subset of the branches above
     branches: [ main ]
 
+permissions:
+  contents: read
+
 jobs:
   analyze:
+    permissions:
+      actions: read  # for github/codeql-action/init to get workflow details
+      contents: read  # for actions/checkout to fetch code
+      security-events: write  # for github/codeql-action/autobuild to send a status report
```

Comment:
Based on the log the security events still seem to have read permissions only. Though the rest was indeed set to none. Is this an upstream bug or I misunderstand something?

```
GITHUB_TOKEN Permissions
  Actions: read
  Contents: read
  Metadata: read
  SecurityEvents: read
  ```

Reply:
AFAIK, the SecurityEvents: write is required for writing events. Thanks!